File: ruby/e10.branch.rb
#!/usr/bin/ruby
print "Input a character: " # prompt
STDOUT.flush                #  flush unbuffered stream
keypress = STDIN.getc       # acquire single character
 
# test if keypress matches pattern
result = case keypress
  when /[a-z]/ then "Lowercase letter"
  when /[A-Z]/ then "Uppercase letter"
  when /[0-9]/ then "Digit"
  else "Punctuation, whitespace, or other"
end
 
puts result

File: perl/e10.branch.pl
#!/usr/bin/perl -w
use v5.10.1;
print "Input a character: "; $keypress=getc(STDIN);
  
# given construction
given ($keypress) {
  when (/[[:lower:]]/) { print "Lowercase letter\n"; }
  when (/[[:upper:]]/) { print "Uppercase letter\n"; }
  when (/[[:digit:]]/) { print "Digit\n" }
  default { print "Punctuation, whitespace, or other\n"; }
}
